examples/step-69/helper.h

```diff
@@ -0,0 +1,83 @@
+#ifndef HELPER_H
+#define HELPER_H
+
+#include <deal.II/base/function.h>
+#include <deal.II/base/tensor.h>
+
+#include <type_traits>
+
+/*
+ * A convenience macro that automatically writes out an accessor (or
+ * getter) function:
+ *
+ *   const Foo& bar() const
+ *   {
+ *      return bar_;
+ *   }
+ *
+ * or
+ *
+ *   const Foo& bar() const
+ *   {
+ *      return *bar_;
+ *   }
+ *
+ * depending on whether bar_ can be dereferenced, or not.
+ */
+
+namespace
+{
+  template <typename T>
+  class is_dereferenciable
```

Comment:
I think this might go into the `template_constraints.h` header file. Then users don't have to worry about it.